detectron2/utils/events.py

```diff
@@ -478,9 +497,282 @@ def clear_images(self):
+            # NOTE: there can be mutliple datasets used together like -('coco', 'voc')
+            # we need to handle each dataset and corresponding table separately 
+            for pred in storage._predictions:
+                loader_i = pred['loader_idx']
+                file_name = pred['file_name']
+                pred_img, avg_bbox_conf = self._plot_prediction(pred)
+
+                # hardcode media panel images upper limit to 8
+                if len(self._media) < 8:
+                    self._media.append(pred_img)
+                if self._table_logging():
+                    # Log images only once, then use their refernces to dedupe
+                    if self._evalset_table[loader_i] is None:
+                        tables[loader_i].add_data(file_name, pred_img, 0, *avg_bbox_conf)
+                        self._map_table_row_file_name[loader_i][file_name] = table_row_idx[loader_i]
+                        table_row_idx[loader_i] = table_row_idx[loader_i] + 1
+                    else:
+                        row_idx = self._map_table_row_file_name[loader_i][file_name]
+                        table_row = self._evalset_table[loader_i].data[row_idx]
+                        tables[loader_i].add_data(
+                            table_row[0],
+                                wandb.Image(table_row[1], boxes=pred_img._boxes, masks=pred_img._masks),
+                                0,
+                                *avg_bbox_conf
+                                )
+
+            if self._table_logging() and storage._predictions:
+                for loader_i, table in enumerate(tables):
+                    table_name = self._dataset[loader_i] + "_"+ str(loader_i)
+                    self._use_table_as_artifact(table, table_name, loader_i)
+                    log_dict[table_name] = table
+
+            log_dict["predictions"] = self._media
+            storage.clear_predictions()
+                    
+        for k, (v, _) in storage.latest_with_smoothing_hint(self._window_size).items():
+            log_dict[k] = v
+
+        self._run.log(log_dict)
+
+    def _use_table_as_artifact(self, table, table_name, loader_i):
+        '''
+        This function logs the given table as artifact and calls `use_artifact` on it so tables from next iter-
+        ations can use the reference of already uploaded images.
+        '''
+        if self._evalset_table[loader_i] is None:
+            eval_art = wandb.Artifact(self._run.id + table_name, type="dataset")
+            eval_art.add(table, table_name)
+            self._run.use_artifact(eval_art)
+            eval_art.wait()
+            self._evalset_table[loader_i] = eval_art.get(table_name)
+
+
+    def _table_logging(self):
+        '''
+        This function returns true if user defined settings enable tables logging implicitly or explicitly
+        '''
+        return True
+
+    def _build_evalset_tables(self):
+        '''
+        
```

Comment:
Probably should add a docstring here for your current preferred design, can be updated as needed

Reply:
The current design is mentioned in line comment. Docstrings is saved to reference docs generation